app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,88 @@
+class TweetsController < ApplicationController
+
+  get "/tweets" do
+    if logged_in?
+      @user = User.find_by(username: session[:username])
+      erb :"/tweets/index"
+    else
+      redirect "/login"
+    end
+  end
+
+
+  get '/tweets/new' do
+    if logged_in?
+      @user = User.find_by(params[:user])
+      erb :"/tweets/new"
+    else
+      redirect :'/login'
+    end
+  end
+
+  get '/tweets/:id' do
+    if logged_in?
+      @tweet = Tweet.find(params[:id])
+      erb :"/tweets/show"
+    else
+      redirect :'/login'
+    end
+  end
+
+  get '/tweets/:id/edit' do
+    if logged_in?
+      @tweet = Tweet.find(params[:id])
+      if @tweet && @tweet.user.username == session[:username]
+        erb :"/tweets/edit"
+      else
+        redirect '/tweets'
+      end
+    else
+      redirect :'/login'
+    end
+  end
+
+  post '/tweets' do
+    if params[:content]==""
+      redirect "/tweets/new"
+    else
+      @tweet = Tweet.create(content: params[:content])
```

Comment:
The tweet creation and user association can both be accomplished in the same step:

``` ruby
@tweet = Tweet.create(content: params[:content], user: current_user)
```